adam-core/src/main/scala/org/bdgenomics/adam/converters/VariantAnnotations.scala

```diff
@@ -0,0 +1,140 @@
+ * limitations under the License.
+ */
+
+package org.bdgenomics.adam.converters
+
+import htsjdk.variant.vcf.VCFConstants
+import htsjdk.variant.variantcontext.VariantContext
+import org.bdgenomics.formats.avro.{ TranscriptEffect, Variant, VariantAnnotation, VariantAnnotationMessage }
+import org.bdgenomics.formats.avro.VariantAnnotationMessage._
+import org.bdgenomics.utils.misc.Logging
+import scala.collection.JavaConverters._
+
+object VariantAnnotations extends Serializable with Logging {
+  private val MESSAGES: Map[String, VariantAnnotationMessage] = Map(
+    // name -> enum
+    ERROR_CHROMOSOME_NOT_FOUND.name() -> ERROR_CHROMOSOME_NOT_FOUND,
+    ERROR_OUT_OF_CHROMOSOME_RANGE.name() -> ERROR_OUT_OF_CHROMOSOME_RANGE,
+    WARNING_REF_DOES_NOT_MATCH_GENOME.name() -> WARNING_REF_DOES_NOT_MATCH_GENOME,
+    WARNING_SEQUENCE_NOT_AVAILABLE.name() -> WARNING_SEQUENCE_NOT_AVAILABLE,
+    WARNING_TRANSCRIPT_INCOMPLETE.name() -> WARNING_TRANSCRIPT_INCOMPLETE,
+    WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS.name() -> WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS,
+    WARNING_TRANSCRIPT_NO_START_CODON.name() -> WARNING_TRANSCRIPT_NO_START_CODON,
+    INFO_REALIGN_3_PRIME.name() -> INFO_REALIGN_3_PRIME,
+    INFO_COMPOUND_ANNOTATION.name() -> INFO_COMPOUND_ANNOTATION,
+    INFO_NON_REFERENCE_ANNOTATION.name() -> INFO_NON_REFERENCE_ANNOTATION,
+    // message code -> enum
+    "E1" -> ERROR_CHROMOSOME_NOT_FOUND,
+    "E2" -> ERROR_OUT_OF_CHROMOSOME_RANGE,
+    "W1" -> WARNING_REF_DOES_NOT_MATCH_GENOME,
+    "W2" -> WARNING_SEQUENCE_NOT_AVAILABLE,
+    "W3" -> WARNING_TRANSCRIPT_INCOMPLETE,
+    "W4" -> WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS,
+    "W5" -> WARNING_TRANSCRIPT_NO_START_CODON,
+    "I1" -> INFO_REALIGN_3_PRIME,
+    "I2" -> INFO_COMPOUND_ANNOTATION,
+    "I3" -> INFO_NON_REFERENCE_ANNOTATION
+  )
+
+  private def parseEffects(s: String): List[String] = {
+    s.split("&").toList
+  }
+
+  private def parseMessages(s: String): List[VariantAnnotationMessage] = {
+    // todo: haven't seen a delimiter here, assuming it is also '&'
+    s.split("&").map(MESSAGES.get(_)).toList.flatten
+  }
+
+  private def parseFraction(s: String): (Option[Integer], Option[Integer]) = {
+    if ("".equals(s)) {
+      return (None, None)
+    }
+    val tokens = s.split("/")
+    tokens.length match {
+      case 0 => (None, None)
+      case 1 => (Some(Integer.parseInt(tokens(0))), None)
+      case _ => (Some(Integer.parseInt(tokens(0))), Some(Integer.parseInt(tokens(1))))
+    }
+  }
+
+  private def notEmpty(s: String): Boolean = Option(s) match {
+    case None    => false
+    case Some(s) => s.length > 0
+  }
+
+  private[converters] def parseTranscriptEffect(s: String): Seq[TranscriptEffect] = {
+    val fields = s.split("\\|", -1)
+    if (fields.length < 16) {
+      if (log.isWarnEnabled) log.warn("Invalid VCF ANN attribute, expected at least 16 fields, found {}: {}", fields.length, fields)
```

Comment:
We don't have log level checks in `o.b.utils.Logging` methods, so this uses `log` directly.

`ValidationStringency` would be nice to have, but would need to be pushed down all the way through VCF parsing, and I'm not sure Hadoop-BAM supports it.